File: Python/QuickSort.py
#QuickSort-Top2
from random import choice
def QuickSort(l):
    #Recursive Key
    if len(l)<=1:
        return l
        
    #Divide list on thre other list
    litle,equal,large=[],[],[]
    pivot=choice(l)
    for a in l:
        if pivot>a:litle.append(a)
        elif pivot<a:large.append(a)
        else:equal.append(a)
    
    #Do recursive
    return QuickSort(litle)+equal+QuickSort(large)

#Create ourList and check QuickSort
ourList=[5,23,65,32,12,34,86,43]
print(QuickSort(ourList))

File: Js/QuickSort.js
const QuickSort = (l) => {
  if (l.length < 2) {
    return l;
  }
  const pivot = l[Math.round(Math.random() * (l.length - 1))];
  const left = l.filter((i) => i < pivot);
  const center = l.filter((i) => i === pivot);
  const right = l.filter((i) => i > pivot);
  return [...QuickSort(left), ...center, ...QuickSort(right)];
};

const l = [2, 5, 1, 7, 3, 9, 43, 12, 13, 13, 54, 23, 11];
console.log(QuickSort(l));
